modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/repository/RepositoryDistributionSnapshot.java

```diff
@@ -434,4 +447,128 @@ public DistributionSnapshot readJson(InputStream in) {
+        final DocumentModel doc = getDoc();
+        if (updateProperties == null) {
+            return doc;
+        }
+        // validations
+        if (Stream.of(TITLE_PROPERTY_PATH, PROP_NAME, PROP_VERSION, PROP_KEY)
+                  .anyMatch(p -> updateProperties.containsKey(p) && StringUtils.isBlank(updateProperties.get(p)))) {
+            throw new DocumentValidationException("Please fill all required fields.");
+        }
+        if (updateProperties.containsKey(PROP_KEY)) {
+            validateKeyOrAlias(updateProperties.get(PROP_KEY), reservedKeys);
+        }
+        List<String> aliases = null;
+        if (updateProperties.containsKey(PROP_ALIASES)) {
+            aliases = Arrays.stream(updateProperties.get(PROP_ALIASES).split("\n"))
+                            .map(String::trim)
+                            .filter(StringUtils::isNotBlank)
+                            .collect(Collectors.toList());
+            aliases.stream().forEach(alias -> validateKeyOrAlias(alias, reservedKeys));
+        }
+        // updates
+        Stream.of(TITLE_PROPERTY_PATH, PROP_NAME, PROP_VERSION, PROP_KEY, PROP_LATEST_LTS, PROP_LATEST_FT, PROP_HIDE)
+              .filter(updateProperties::containsKey)
+              .forEach(p -> doc.setPropertyValue(p, updateProperties.get(p)));
+        Stream.of(PROP_LATEST_LTS, PROP_LATEST_FT, PROP_HIDE)
+              .forEach(p -> doc.setPropertyValue(p, updateProperties.get(p)));
+        if (updateProperties.containsKey(PROP_RELEASED)) {
+            doc.setPropertyValue(DistributionSnapshot.PROP_RELEASED, convertDate(updateProperties.get(PROP_RELEASED)));
+        }
+        if (aliases != null) {
+            doc.setPropertyValue(PROP_ALIASES, (Serializable) aliases);
+        }
+        if (!StringUtils.isBlank(comment)) {
+            doc.putContextData("comment", comment);
+        }
+        doc.putContextData(ThumbnailConstants.DISABLE_THUMBNAIL_COMPUTATION, true);
+        DocumentModel updatedDoc = session.saveDocument(doc);
+        session.save();
+        return updatedDoc;
+    }
+
+    protected void validateKeyOrAlias(String keyOrAlias, List<String> reservedKeys) throws DocumentValidationException {
+        var forbidden = new ArrayList<>(List.of(
+                // reserved for live distrib
+                SnapshotManager.DISTRIBUTION_ALIAS_CURRENT, SnapshotManager.DISTRIBUTION_ALIAS_ADM,
+                // added automatically
+                SnapshotManager.DISTRIBUTION_ALIAS_LATEST_FT, SnapshotManager.DISTRIBUTION_ALIAS_LATEST_LTS));
+        if (reservedKeys != null) {
+            forbidden.addAll(reservedKeys);
+        }
+        Optional<String> match = forbidden.stream().filter(keyOrAlias::equals).findFirst();
+        if (match.isPresent()) {
+            throw new DocumentValidationException(
+                    String.format("Distribution key or alias is reserved: '%s'", match.get()));
+        }
```

Comment:
Why not simply `if (forbidden.contains(keyOrAlias))`?

Reply:
Indeed, maybe i contracted some kind of lambda-related fever... :+1: